rfcs/0016-verification/0016-verification.md

```diff
@@ -0,0 +1,117 @@
+---
+Number: "0016"
+Category: Informational
+Status: Draft
+Author: Dingwei Zhang
+Organization: Nervos Foundation
+Created: 2019-03-08
+---
+
+# CKB-Verification
+
+## Abstract
+
+This RFC describes ckb block data verification rules.
+
+Verification rules are split into three part in CKB implementation. `Header Verification`,  `Block Verification` and `Transaction Verification`.
+
+## Specification
+
+1. Header Verification
+
+    1. proof-of-work verification
+
+        Block to be valid it must hash to a value less than the current target. current hash algorithm is Cuckoo Cycle.
+
+    2. number verification
+
+        CKB block header contain block number, it MUST equal parent’s block number incremented by one, the genesis block has a number of zero.
+
+    3. block version verification
+
+        Block version MUST obey consensus rules.
+
+    4. timestamp verification
+
+        Timestamp (Unix’s time in milliseconds) of block timestamp MUST greater than the median timestamp of previous 11 blocks and less than the network-adjusted time + 15 seconds.
+
+    5. difficulty verification
+
+        The canonical difficulty of a block of header is defined as Diff == Max(Diff_last * o_last / o, Diff_last * 2) (o: orphan rate), It MUST greater or equal genesis' difficulty.
+
+1. Block Verification
+
+   1. empty transactions verification
```

Comment:
I think this rule is redundant with cellbase rule (block MUST include exactly one cellbase transaction).